File: DoohAdManager.Model/Application/DBO/CampaignMedia.cs
namespace DoohAdManager.Model.Application.DBO
{
    public class MvCampaignMedia
    {
        public int CampaignId { get; set; }
        public int ScreenId { get; set; }
        public string? ScreenName { get; set; }
        public List<MvMediaItem>? Media { get; set; }
        public DateOnly PlayDate { get; set; }
        public DateTime CreatedAt { get; set; }
        public int CreatedBy { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public int? UpdatedBy { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? DeletedAt { get; set; }
        public int? DeletedBy { get; set; }
    }

    public class MvMediaItem
    {
        public int Id { get; set; }
        public int MediaId { get; set; }
        public string? MediaName { get; set; }
        public bool MediaType { get; set; }
        public string? MediaUrl { get; set; }
        public int PlaySequence { get; set; }
    }

    public class MvCampaignMediaAdd
    {
        public int CampaignId { get; set; }
        public int ScreenId { get; set; }
        public DateOnly PlayDate { get; set; }
        public int CreatedBy { get; set; }
        public required List<MvMediaItemAdd> Media { get; set; }
    }

    public class MvMediaItemAdd
    {
        public int MediaId { get; set; }
        public int PlaySequence { get; set; }
    }

    public class MvCampaignMediaUpdate
    {
        public int CampaignId { get; set; }
        public int ScreenId { get; set; }
        public DateOnly PlayDate { get; set; }
        public int UpdatedBy { get; set; }
        public required List<MvMediaItemUpdate> Media { get; set; }
    }

    public class MvMediaItemUpdate
    {
        public int Id { get; set; }
        public int PlaySequence { get; set; }
    }

    public class MvCampaignMediaFilter
    {
        public int CampaignId { get; set; }
        public int? ScreenId { get; set; }
        public DateOnly? PlayDate { get; set; }
        public string? Search { get; set; }
    }

    public class MvCampaignMediaDelete
    {
        public int Id { get; set; }
        public int DeletedBy { get; set; }
    }

    public class MvCampaignMediaDeleted
    {
        public int Id { get; set; }
        public int CampaignId { get; set; }
        public int ScreenId { get; set; }
        public int MediaId { get; set; }
        public DateOnly PlayDate { get; set; }
        public int PlaySequence { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime CreatedAt { get; set; }
        public int CreatedBy { get; set; }
        public DateTime? DeletedAt { get; set; }
        public int? DeletedBy { get; set; }
    }
}

File: dooh-ad-manager/src/app/Application/DBO/Campaign/Models/campaign-media.ts
export interface MvCampaignMedia {
  campaignId: number;
  screenId: number;
  screenName: string;
  media: MvMediaItem[];
  playDate: string;
  createdAt: Date;
  updatedAt: Date;
  updatedBy: number;
  isDeleted: boolean;
  deletedAt: Date;
  deletedBy: number;
}

export interface MvMediaItem {
  id: number;
  mediaId: number;
  mediaName: string;
  mediaType: string;
  mediaUrl: string;
  playSequence: number;
}

export class MvCampaignMediaAdd {
  campaignId: number;
  screenId: number;
  playDate: string;
  createdBy: number;
  media: MvMediaItemAdd[];

  constructor() {
    this.campaignId = 0;
    this.screenId = 0;
    this.playDate = '';
    this.createdBy = 1;
    this.media = [];
  }
}

export interface MvMediaItemAdd {
  mediaId: number;
  playSequence: number;
}

export class MvCampaignMediaUpdate {
  campaignId: number;
  screenId: number;
  playDate: string;
  updatedBy: number;
  media: MvMediaItemUpdate[];

  constructor() {
    this.campaignId = 0;
    this.screenId = 0;
    this.playDate = '';
    this.updatedBy = 1;
    this.media = [];
  }
}

export interface MvMediaItemUpdate {
  id: number;
  playSequence: number;
}

export interface MvCampaignMediaFilter {
  campaignId: number;
  screenId?: number;
  playDate?: string | null;
  search?: string;
}

export interface MvCampaignMediaDelete {
  id: number;
  deletedBy: number;
}

export interface MvCampaignMediaDeleted {
  id: number;
  campaignId: number;
  screenId: number;
  mediaId: number;
  playDate: string;
  playSequence: number;
  isDeleted: boolean;
  createdAt: Date;
  createdBy: number;
  deletedAt?: Date;
  deletedBy?: number;
}
